### PR TITLE
Feat: AI proxy advanced

### PR DESCRIPTION
**Pull request by lena-larionova**
### Description

AI Proxy Advanced plugin.

Schema (config reference) and basic example added via https://github.com/Kong/docs-plugin-toolkit/pull/47.

https://konghq.atlassian.net/browse/DOCU-3884

Questions for @fffonion @AntoineJac:
* How many of the currently existing [AI Proxy docs](https://docs.konghq.com/hub/kong-inc/ai-proxy/how-to/llm-provider-integration-guides/anthropic/), especially the LLM integration guides, do we want to reuse for this plugin? How many of them apply?
* Is there somewhere that I can find info on the load balancing and semantic routing capabilities of this plugin?
  * Is the plugin's `lowest-usage` algorithm analogous to the Gateway `least-connections` algorithm defined here: https://docs.konghq.com/gateway/latest/how-kong-works/load-balancing/#least-connections
  * Are the `consistent-hashing` and `round-robin` algorithms that we already use in the Gateway exactly the same as we already define on that same page?

### Testing instructions

Preview link: https://deploy-preview-7817--kongdocs.netlify.app/hub/kong-inc/ai-proxy-advanced/unreleased/

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

File: app/_hub/kong-inc/ai-proxy-advanced/overview/_index.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+## Load balancing
+
+This plugin supports following load balancing algorithms:
+* Lowest-usage 
+* [Round-robin (weighted)](/gateway/latest/how-kong-works/load-balancing/#round-robin)
+* [Consistent-hashing (sticky-session on given header value)](/gateway/latest/how-kong-works/load-balancing/#consistent-hashing)
+
+## Semantic routing
+
+Info about semantic routing?
```
Comment (fffonion) [inline app/_hub/kong-inc/ai-proxy-advanced/overview/_index.md:53]: This plugin supports distributing requests based on similarity of prompts to the given description of each model, allowing Kong to select specific model that performs better
in given domains.

**Comment by fffonion** [comment]
> How many of the currently existing [AI Proxy docs](https://docs.konghq.com/hub/kong-inc/ai-proxy/how-to/llm-provider-integration-guides/anthropic/), especially the LLM integration guides, do we want to reuse for this plugin? How many of them apply?

All existing configurations, guides etc of ai-proxy can be reused for ai-proxy-advanced. It's sharing the same codebase, thus I think we can share the docs too.


> Is there somewhere that I can find info on the load balancing and semantic routing capabilities of this plugin?
Is the plugin's lowest-usage algorithm analogous to the Gateway least-connections algorithm defined here: https://docs.konghq.com/gateway/latest/how-kong-works/load-balancing/#least-connections
Are the consistent-hashing and round-robin algorithms that we already use in the Gateway exactly the same as we already define on that same page?

Yes, the load-balancing algorithms are same thing as the algorithms for kong upstreams. Semantic routing is actually similar as lowest-usage or least-connection, given that the datapoint is the similiary score vs. usage volume or connection count.

**Comment by Guaris** [comment]
@fffonion Hey! I added all of the docs from the AI proxy plugin to the AI proxy advanced plugin section and renamed them. 
https://deploy-preview-7817--kongdocs.netlify.app/
Can you confirm that the following docs still apply with my changes?
https://deploy-preview-7817--kongdocs.netlify.app//hub/kong-inc/ai-proxy-advanced/unreleased/how-to/streaming/
https://deploy-preview-7817--kongdocs.netlify.app//hub/kong-inc/ai-proxy-advanced/unreleased/how-to/sdk-usage/
https://deploy-preview-7817--kongdocs.netlify.app//hub/kong-inc/ai-proxy-advanced/unreleased/how-to/cloud-provider-authentication/

And in, the mistral integration doc for AI proxy we reference this file https://github.com/kong/kong/blob/master/spec/fixtures/ai-proxy/oas.yaml

Is there a corresponding filre for AI Proxy Advanced?

**Comment by fffonion** [comment]
@Guaris thanks! one fix is needed, in `ai-proxy-advanced`, we need to update the config path of `config.model `to config.targets[].model`.
For example:
```
curl -X POST http://localhost:8001/routes/{routeName|Id}/plugins \
    --header "accept: application/json" \
    --header "Content-Type: application/json" \
    --data '
    {
  "name": "ai-proxy-advanced",
  "config": {
    "route_type": "llm/v1/chat",
    "auth": {
      "azure_use_managed_identity": true
    },
    "model": {
      "provider": "azure",
      "name": "gpt-35-turbo",
      "options": {
        "azure_instance": "my-openai-instance",
        "azure_deployment_id": "kong-gpt-3-5"
      }
    }
  }
}
```
would become
```
curl -X POST http://localhost:8001/routes/{routeName|Id}/plugins \
    --header "accept: application/json" \
    --header "Content-Type: application/json" \
    --data '
   {
    "name": "ai-proxy-advanced",
    "config": {
      "targets": [
        {
          "route_type": "llm/v1/chat",
          "auth": {
            "azure_use_managed_identity": true
          },
          "model": {
            "provider": "azure",
            "name": "gpt-35-turbo",
            "options": {
              "azure_instance": "my-openai-instance",
              "azure_deployment_id": "kong-gpt-3-5"
            }
          }
        }
      ]
    }
  }
```

**Comment by Guaris** [comment]
@fffonion 
Isn't that what appears on this page?
https://deploy-preview-7817--kongdocs.netlify.app/hub/kong-inc/ai-proxy-advanced/unreleased/how-to/basic-example/

```
curl -X POST http://localhost:8001/routes/{routeName|Id}/plugins \
   --header "accept: application/json" \
   --header "Content-Type: application/json" \
   --data '
   {
 "name": "ai-proxy-advanced",
 "config": {
   "targets": [
     {
       "route_type": "llm/v1/chat",
       "auth": {
         "header_name": "Authorization",
         "header_value": "Bearer token"
       },
       "model": {
         "provider": "openai"
       }
     },
     {
       "route_type": "llm/v1/chat",
       "model": {
         "provider": "llama2",
         "options": {
           "llama2_format": "openai",
           "upstream_url": "http://httpbin"
         }
       }
     }
   ]
 }
}
   '
```

**Review by Guaris — APPROVED** [review]

**Pull request merged**